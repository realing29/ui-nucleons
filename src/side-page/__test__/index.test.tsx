import React from 'react';
import { render } from '@testing-library/react';
import { Simulate } from 'react-dom/test-utils';
import { SidePage } from '..';

describe('SidePage', () => {
  it('should renders correctly', async function () {
    const { container, rerender } = render(
      <SidePage withTransitions shown={false}>
        <SidePage.Header title='Test title' />
        <SidePage.Body>
          <h1>Test body</h1>
        </SidePage.Body>
        <SidePage.Footer>
          <p>Test footer</p>
        </SidePage.Footer>
      </SidePage>,
    );

    expect(container).toMatchSnapshot();

    rerender(
      <SidePage withTransitions shown>
        <SidePage.Header title='Test title' />
        <SidePage.Body>
          <h1>Test body</h1>
        </SidePage.Body>
        <SidePage.Footer>
          <p>Test footer</p>
        </SidePage.Footer>
      </SidePage>,
    );

    // ждем окончания анимации
    await new Promise(done => setTimeout(done, 1000));

    expect(container).toMatchSnapshot();
    expect(container.querySelectorAll('.enter-active')).toHaveLength(0);
  });

  it('should handle header props', async function () {
    const onBack = jest.fn();
    const onClose = jest.fn();

    const { getByTestId } = render(
      <SidePage shown>
        <SidePage.Header title='Test title' onBack={onBack} onClose={onClose} />
        <SidePage.Body>
          <h1>Test body</h1>
        </SidePage.Body>
        <SidePage.Footer>
          <p>Test footer</p>
        </SidePage.Footer>
      </SidePage>,
    );

    expect(onBack).toBeCalledTimes(0);
    Simulate.click(getByTestId('side-page:back'));
    expect(onBack).toBeCalledTimes(1);

    expect(onClose).toBeCalledTimes(0);
    Simulate.click(getByTestId('side-page:close'));
    expect(onClose).toBeCalledTimes(1);
  });
});
