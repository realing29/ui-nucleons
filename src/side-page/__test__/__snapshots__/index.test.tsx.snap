// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePage should renders correctly 1`] = `<div />`;

exports[`SidePage should renders correctly 2`] = `
<div>
  <div
    class="root overlay"
    data-testid="modal-overlay"
    style="--transition-duration: 300ms; z-index: 100; --vh: 7.68px;"
  >
    <div
      class="main size-m"
    >
      <div
        class="root size-xl header no-subtitle"
        data-testid="top-bar"
        style="height: 80px;"
      >
        <div
          class="main"
        >
          <div
            class="title"
          >
            Test title
          </div>
        </div>
      </div>
      <div
        class="body"
      >
        <h1>
          Test body
        </h1>
      </div>
      <div
        class="root rounds-md__b footer padded"
        data-testid="bottom-bar"
        style="height: 80px;"
      >
        <p>
          Test footer
        </p>
      </div>
    </div>
  </div>
</div>
`;
